client.go

```diff
@@ -197,6 +202,63 @@ func (e *FailedResponse) Error() string {
 	return fmt.Sprintf("%v: %s", e.StatusCode, string(e.Response))
 }
 
+// Patch updates a resource for the target URL
+func (c *Client) Patch(url string, contentType ContentType, body interface{}, dest interface{}) error {
+	var err error
+	jsonBody, err := json.Marshal(body)
+
+	if err != nil {
+		return fmt.Errorf("could not patch, body could not be marshalled to json: %w", err)
+	}
+
+	req, err := http.NewRequest("PATCH", url, bytes.NewBuffer(jsonBody))
+
+	if err != nil {
+		return err
+	}
+
+	if c.FiatUser != "" {
+		req.Header.Set(SpinFiatUserHeader, c.FiatUser)
+		req.Header.Set(SpinFiatAccountHeader, c.FiatUser)
+	}
+
+	req.Header.Set("Content-Type", string(contentType))
+
+	resp, err := c.http.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	b, err := ioutil.ReadAll(resp.Body)
```

Comment:
Ideally would only read the body if the status code >= 200  <400 (really <300).
If there's a different status code, the idiomatic way is to `io.Copy(ioutil.Discard, resp.Body)` which doesn't allocate a new byte array.

Reply:
This is copied from existing methods in plank heh (done out of expedience for the experiment). Ideally we have all this logic in a central place, since there's little difference in the implementations. I'll fiddle with it this evening and see what we can come up with!